update docs to promote PodReadyToStartContainersCondition into beta (#43533)
kubernetes/website: content/en/docs/concepts/workloads/pods/pod-lifecycle.md

PodConditions:

* `PodScheduled`: the Pod has been scheduled to a node.

> * `PodReadyToStartContainers`: (beta feature; enable by [default now](#pod-has-network)) the

Reviewer: ```suggestion
* `PodReadyToStartContainers`: (beta feature; enabled by default) the
```

### Pod network readiness {#pod-has-network}

{{< feature-state for_k8s_version="v1.29" state="beta" >}}

{{< note >}}

> This condition was renamed from PodHasNetwork to PodReadyToStartContainers. And now is enable by default

Reviewer: ```suggestion
During its early development, this condition was named `PodHasNetwork`.
```

Author: updated

{{< /note >}}

> After a Pod gets scheduled on a node, it needs to be admitted by the Kubelet and
> have any volumes mounted. Once these phases are complete, the Kubelet works with

Reviewer: ```suggestion
After a Pod gets scheduled on a node, it needs to be admitted by the kubelet and
to have any required storage volumes mounted. Once these phases are complete,
the kubelet works with
```

Author: updated

a container runtime (using {{< glossary_tooltip term_id="cri" >}}) to set up a

> runtime sandbox and configure networking for the Pod,

Reviewer: ```suggestion
runtime sandbox and configure networking for the Pod. If the
`PodReadyToStartContainersCondition`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled
(it is enabled by default for Kubernetes {{< skew currentVersion >}}), the
```

> Kubelet reports whether a pod has reached this initialization milestone through

Reviewer: ```suggestion
kubelet reports whether a pod has reached this initialization milestone through
```

Author: updated

> ### Pod network readiness {#pod-has-network}

Reviewer 2: @sftim wdyt about changing the name of this section? This was the name of the old condition and we renamed it to PodReadyToStartContainers.

Reviewer: How about “Pod sandbox readiness”, then?

Reviewer: We can make that change after this PR lands.

{{< feature-state for_k8s_version="v1.29" state="beta" >}}

{{< note >}}
During its early development, this condition was named `PodHasNetwork`.
{{< /note >}}

After a Pod gets scheduled on a node, it needs to be admitted by the kubelet and
to have any required storage volumes mounted. Once these phases are complete,
the kubelet works with
a container runtime (using {{< glossary_tooltip term_id="cri" >}}) to set up a
runtime sandbox and configure networking for the Pod. If the
`PodReadyToStartContainersCondition`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled
(it is enabled by default for Kubernetes {{< skew currentVersion >}}), the

> `PodReadyToStartContainers` condition in the `status.conditions` field of a Pod.

Reviewer 2: ```suggestion
`PodReadyToStartContainers` condition will be added to the `status.conditions` field of a Pod.
```

Author: updated
